### PR TITLE
feat: add reusable upload component to storybook #1095

### LINKED ISSUE #1095: Add Upload to Storybook and make it reusable and move it into it's own component
We introduced a new Feedback widget width a drag'n'drop file upload. We would like to make this component reusable in other pages and add it to our Storybook.

Figma designs are [here](https://www.figma.com/file/eq3JgvmW48IVrYcg2kzgLL/Komiser---Design-System?type=design&node-id=2328-15447&mode=dev)

Screenshot 2023-10-17 at 10 26 50

You will find the current upload form inside `/Users/traxmaxx/Sites/tailwarden/komiser/dashboard/components/feedback-widget/FeedbackWidget.tsx:1`. Look for something like

```typescript
 <FileUploader
                    classes={
                      fileAttachement === null
                        ? `grow cursor-pointer rounded border-2 border-dashed border-black-170 py-5 text-center text-xs transition hover:border-[#B6EAEA] hover:bg-black-100 w-[50%]`
                        : `grow min-h-full flex-1 rounded border-2 border-[#B6EAEA] text-center text-xs transition w-full`
                    }
                    disabled={
                      fileAttachement !== null ||
                      isSendingFeedback ||
                      isTakingScreenCapture
                    }
                    handleChange={uploadFile}
                    name="attachement"
                    types={FILE_TYPES}
                    fileOrFiles={fileAttachement}
                    hoverTitle="drop here"
                    maxSize={MAX_FILE_SIZE_MB}
                    onTypeError={(err: string) =>
                      setToast({
                        hasError: true,
                        title: 'File upload failed',
                        message: err
                      })
                    }
                    onSizeError={(err: string) =>
                      setToast({
                        hasError: true,
                        title: 'File upload failed',
                        message: err
                      })
                    }
                    dropMessageStyle={{
                      width: '100%',
                      height: '100%',
                      position: 'absolute',
                      background: '#F4F9F9',
                      top: 0,
                      right: 2,
                      display: 'flex',
                      flexGrow: 2,
                      opacity: 1,
                      zIndex: 20,
                      color: '#008484',
                      fontSize: 14,
                      border: 'none'
                    }}
                  >
...
 </FileUploader>
```

Please move this into it's own component with all necessary props and stylings and make sure the FeedbackWidget is using the new implementation. Also please add the new component to Storybook with all necessary variants and a proper documentation.

**Additional context**
If you want to tackle this task, feel free to share with us your email address so we can give you access to the Figma project :)

### PR DESCRIPTION
**Pull request by faisal7008**
## Problem

The existing Feedback widget in our application had a file upload feature that wasn’t reusable and wasn’t included in our Storybook.

fixes #1095 

## Solution

To address this issue, I refactored the file upload feature into its own Upload component. This new component includes all the necessary props and stylings, making it easily reusable across different parts of our application. I also ensured that the FeedbackWidget is now using this new Upload component. In addition, I added the Upload component to our Storybook. This includes all necessary variants (supporting both single and multiple file uploads)

## Changes Made

- Refactored file upload feature into a new, reusable Upload component.
- Updated FeedbackWidget to use the new Upload component.
- Added Upload component to Storybook with support for both single and multiple file uploads.

## Screenshots

Screenshot 2023-10-19 at 8 41 57 PM

Screenshot 2023-10-19 at 8 42 22 PM

## Checklist

- [x] Code follows the <a href="https://github.com/tailwarden/komiser/blob/master/CONTRIBUTING.md">contributing</a> guidelines
- [x] Changes have been thoroughly tested
- [x] <a href="https://github.com/tailwarden/docs.komiser.io">Documentation</a> has been updated, if necessary
- [x] Any dependencies have been added to the project, if necessary

## Reviewers

@Traxmaxx

### CODE REVIEW

**Comment by Traxmaxx** [comment]
Really great PR so far. Thanks for taking care of the issue! 🙏

**Review by mlabouardy — APPROVED** [review]

File: dashboard/components/feedback-widget/FeedbackWidget.tsx [hunk 1]
```diff
@@ -249,189 +247,32 @@ const useFeedbackWidget = (defaultState: boolean = false) => {
...
-                        <p>
-                          Drag and drop or{' '}
-                          <span className="cursor-pointer text-primary">
-                            choose a file
-                          </span>
-                        </p>
-                      </div>
-                    )}
-                    {fileAttachement !== null && (
-                      <div className="relative h-full w-full px-6 py-5">
```
[the remark this answers is not in the document]
Comment (faisal7008) [reply dashboard/components/feedback-widget/FeedbackWidget.tsx:347]: yes, we can do that. should i make changes and raise PR

**Comment by mlabouardy** [comment]
@faisal7008 can you fix the conflict so the PR can be merged please

**Comment by faisal7008** [comment]
> @faisal7008 can you fix the conflict so the PR can be merged please
fixed the conflict, you can merge the PR now

**Pull request merged**